frontend/src/metabase/visualizations/echarts/cartesian/option/series.ts

```diff
@@ -0,0 +1,159 @@
+    settings["stackable.stack_type"] != null ? `bar_${yAxisIndex}` : undefined;
+
+  return {
+    type: "bar",
+    yAxisIndex,
+    stack: stackName,
+    encode: {
+      y: seriesModel.dataKey,
+      x: dimensionDataKey,
+    },
+    label: buildEChartsLabelOptions(seriesModel, settings, renderingContext),
+    labelLayout: {
+      hideOverlap: settings["graph.label_value_frequency"] === "fit",
+    },
+    itemStyle: {
+      color: seriesColor,
+    },
+  };
+};
+
+const buildEChartsLineAreaSeries = (
+  seriesModel: SeriesModel,
+  seriesSettings: SeriesSettings,
+  seriesColor: string,
+  settings: ComputedVisualizationSettings,
+  dimensionDataKey: string,
+  yAxisIndex: number,
+  renderingContext: RenderingContext,
+): RegisteredSeriesOption["line"] => {
+  const display = seriesSettings?.display ?? "line";
+
+  const stackName =
+    settings["stackable.stack_type"] != null ? `area_${yAxisIndex}` : undefined;
+
+  return {
+    type: "line",
+    yAxisIndex,
+    showSymbol: seriesSettings["line.marker_enabled"] !== false,
+    symbolSize: 6,
+    smooth: seriesSettings["line.interpolate"] === "cardinal",
+    connectNulls: seriesSettings["line.missing"] === "interpolate",
+    step:
+      seriesSettings["line.interpolate"] === "step-after" ? "end" : undefined,
+    stack: stackName,
+    areaStyle: display === "area" ? { opacity: 0.3 } : undefined,
+    encode: {
+      y: seriesModel.dataKey,
+      x: dimensionDataKey,
+    },
+    label: buildEChartsLabelOptions(seriesModel, settings, renderingContext),
+    labelLayout: {
+      hideOverlap: settings["graph.label_value_frequency"] === "fit",
+    },
+    itemStyle: {
+      color: seriesColor,
+    },
+  };
+};
+
+export const buildEChartsSeries = (
+  chartModel: CartesianChartModel,
+  settings: ComputedVisualizationSettings,
+  renderingContext: RenderingContext,
+): EChartsOption["series"] => {
+  return chartModel.seriesModels
+    .map(seriesModel => {
+      const seriesSettings: SeriesSettings = settings.series(
+        seriesModel.legacySeriesSettingsObjectKey,
+      );
+      const seriesColor =
+        settings?.["series_settings.colors"]?.[seriesModel.vizSettingsKey];
+
+      const yAxisIndex = chartModel.yAxisSplit[0].includes(seriesModel.dataKey)
+        ? 0
+        : 1;
+
+      switch (seriesSettings.display) {
+        case "line":
+        case "area":
+          return buildEChartsLineAreaSeries(
+            seriesModel,
+            seriesSettings,
+            seriesColor,
+            settings,
+            chartModel.dimensionModel.dataKey,
+            yAxisIndex,
+            renderingContext,
+          );
+        case "bar":
+          return buildEChartsBarSeries(
+            seriesModel,
+            seriesSettings,
+            seriesColor,
+            settings,
+            chartModel.dimensionModel.dataKey,
+            yAxisIndex,
+            renderingContext,
+          );
+      }
+    })
+    .filter(isNotNull);
```

Comment:
Why do we need the `.filter`? Is it possible that one of the `seriesModels` is null?